klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Run GlobalFoundries 180nm MCU DRC.
+
+Usage:
+    run_drc.py (--help| -h)
+    run_drc.py (--path=<file_path>) (--variant=<combined_options>) [--verbose] [--table=<table_name>]... [--mp=<num_cores>] [--run_dir=<run_dir_path>] [--topcell=<topcell_name>] [--thr=<thr>] [--run_mode=<run_mode>] [--no_feol] [--no_beol] [--connectivity] [--density] [--density_only] [--antenna] [--antenna_only] [--no_offgrid]
+
+Options:
+    --help -h                           Print this help message.
+    --path=<file_path>                  The input GDS file path.
+    --variant=<combined_options>        Select combined options of metal_top, mim_option, and metal_level. Allowed values (A, B, C).
+                                        variant=A: Select  metal_top=30K  mim_option=A  metal_level=3LM
+                                        variant=B: Select  metal_top=11K  mim_option=B  metal_level=4LM
+                                        variant=C: Select  metal_top=9K   mim_option=B  metal_level=5LM
+    --topcell=<topcell_name>            Topcell name to use.
+    --table=<table_name>                Table name to use to run the rule deck.
+    --mp=<num_cores>                    Run the rule deck in parts in parallel to speed up the run. [default: 1]
+    --run_dir=<run_dir_path>            Run directory to save all the results [default: pwd]
+    --thr=<thr>                         The number of threads used in run.
+    --run_mode=<run_mode>               Select klayout mode Allowed modes (flat , deep, tiling). [default: flat]
+    --no_feol                           Turn off FEOL rules from running.
```

Comment:
curious if using "negative" flags there is intentional (it can get confusing when you mix both semantic in a command line application).

Reply:
The use of the flags is to default to what is required for manufacturing and important for the layout engineer at the design time. There are some rules that are important at the end, at tapeout time. FEOL and BEOL are always important. But let's say I'm doing digital cell, I might be interested in BEOL only. I could turn off FEOL.

Reply:
yes, but maybe that could be encoded by documenting the default rather than mixing negative / positive form potion, see http://docopt.org/#:~:text=and%20the%20default%20value%20for%20an%20option%27s%20argument.

Reply:
#20